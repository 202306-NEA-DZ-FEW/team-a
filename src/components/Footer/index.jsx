import Link from "next/link";
import { useTranslation } from "next-i18next";
import { AiFillFacebook } from "react-icons/ai";
import { AiFillLinkedin } from "react-icons/ai";
import { BsShareFill } from "react-icons/bs";
import { FaInstagram } from "react-icons/fa";
import { TbLanguage } from "react-icons/tb";

function Footer() {
  const { t } = useTranslation();
  return (
    <div className='bg-neutral py-10 px-6 text-white w-full'>
      <div className='grid grid-cols-1 justify-items-center md:grid-cols-5 gap-4'>
        <div className='col-span-1'>
          <Link href='/'>
            <h2 className='font-bold text-2xl text-white'>
              {t("common:buttons:logo")}
            </h2>
          </Link>
        </div>
        <div className='col-span-1 '>
          <ul className='flex flex-col items-center md:items-start gap-4'>
            <Link href='/about'>
              <li className='hover:underline'>{t("common:buttons:about")}</li>
            </Link>
            <Link href='/blogs'>
              <li className='hover:underline'>{t("common:buttons:blogs")}</li>
            </Link>
            <Link href='/products'>
              <li className='hover:underline'>
                {t("common:footer:listedItems")}
              </li>
            </Link>
          </ul>
        </div>
        <div className='col-span-1'>
          <ul className='flex flex-col items-center gap-4'>
            <li className=''>{t("common:footer:contactUs")}</li>
            <div className='flex gap-3 text-3xl mt-2 '>
              <Link href='https://www.facebook.com' target='_blank'>
                <AiFillFacebook className='opacity-70 hover:opacity-100' />
              </Link>
              <Link href='https://www.instagram.com' target='_blank'>
                <FaInstagram className='opacity-70 hover:opacity-100' />
              </Link>
              <Link href='https://www.linkedin.com' target='_blank'>
                <AiFillLinkedin className='opacity-70 hover:opacity-100' />
              </Link>
              <Link href='https://www.google.com' target='_blank'>
                <BsShareFill className='opacity-70 hover:opacity-100' />
              </Link>
            </div>
          </ul>
        </div>
        <div className='col-span-1'>
          <ul className='flex flex-col items-center md:items-start'>
            <li className=''>{t("common:footer:address")}</li>
            <li className=''>{t("common:footer:street")}</li>
            <li className=''>{t("common:footer:city")}</li>
          </ul>
        </div>
        <div className='col-span-1'>
          <div className='dropdown dropdown-top dropdown-end'>
            <label
              tabIndex={0}
              className='btn hover:bg-transparent  bg-neutral text-white'
            >
              <TbLanguage className='text-xl' />
            </label>
            <ul
              tabIndex={0}
              className='dropdown-content z-[1] menu text-black shadow bg-white rounded-md w-36'
            >
              <li>
                <Link href='/' locale='en' className='block px-4 py-2 '>
                  <span className='fi fi-gb'></span> English
                </Link>
              </li>
              <li>
                <Link href='/' locale='fr' className='block px-4 py-2'>
                  <span className='fi fi-fr'></span> Français
                </Link>
              </li>
              <li>
                <Link href='/' locale='ar' className='block px-4 py-2 '>
                  <span className='fi fi-sa '></span> العربية
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Footer;
