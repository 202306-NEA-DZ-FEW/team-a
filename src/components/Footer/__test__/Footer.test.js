import renderer from "react-test-renderer";

import Footer from "..";

it("renders correctly", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
