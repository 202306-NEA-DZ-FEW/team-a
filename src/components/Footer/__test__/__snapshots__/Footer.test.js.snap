// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-neutral py-10 px-6 text-white w-full"
>
  <div
    className="grid grid-cols-1 justify-items-center md:grid-cols-5 gap-4"
  >
    <div
      className="col-span-1"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <h2
          className="font-bold text-2xl text-white"
        >
          buttons.logo
        </h2>
      </a>
    </div>
    <div
      className="col-span-1 "
    >
      <ul
        className="flex flex-col items-center md:items-start gap-4"
      >
        <a
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <li
            className="hover:underline"
          >
            buttons.about
          </li>
        </a>
        <a
          href="/blogs"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <li
            className="hover:underline"
          >
            buttons.blogs
          </li>
        </a>
        <a
          href="/items"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <li
            className="hover:underline"
          >
            footer.listedItems
          </li>
        </a>
      </ul>
    </div>
    <div
      className="col-span-1"
    >
      <ul
        className="flex flex-col items-center gap-4"
      >
        <li
          className=""
        >
          footer.contactUs
        </li>
        <div
          className="flex gap-3 text-3xl mt-2 "
        >
          <a
            href="https://www.facebook.com"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            target="_blank"
          >
            <svg
              className="opacity-70 hover:opacity-100"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
              />
            </svg>
          </a>
          <a
            href="https://www.instagram.com"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            target="_blank"
          >
            <svg
              className="opacity-70 hover:opacity-100"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
          <a
            href="https://www.linkedin.com"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            target="_blank"
          >
            <svg
              className="opacity-70 hover:opacity-100"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
              />
            </svg>
          </a>
          <a
            href="https://www.google.com"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            target="_blank"
          >
            <svg
              className="opacity-70 hover:opacity-100"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 2.5a2.5 2.5 0 1 1 .603 1.628l-6.718 3.12a2.499 2.499 0 0 1 0 1.504l6.718 3.12a2.5 2.5 0 1 1-.488.876l-6.718-3.12a2.5 2.5 0 1 1 0-3.256l6.718-3.12A2.5 2.5 0 0 1 11 2.5z"
              />
            </svg>
          </a>
        </div>
      </ul>
    </div>
    <div
      className="col-span-1"
    >
      <ul
        className="flex flex-col items-center md:items-start"
      >
        <li
          className=""
        >
          footer.address
        </li>
        <li
          className=""
        >
          footer.street
        </li>
        <li
          className=""
        >
          footer.city
        </li>
      </ul>
    </div>
    <div
      className="col-span-1"
    >
      <div
        className="relative text-left block lg:block"
        data-headlessui-state=""
      >
        <div>
          <button
            aria-expanded={false}
            aria-haspopup="menu"
            className="btn btn-sm outline-dashed transition-all duration-500 ease-in-out hover:outline-double outline-[2px] btn-circle btn-ghost"
            data-headlessui-state=""
            id="headlessui-menu-button-:r0:"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <svg
              className="w-6 h-6"
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M4 5h7"
              />
              <path
                d="M9 3v2c0 4.418 -2.239 8 -5 8"
              />
              <path
                d="M5 9c0 2.144 2.952 3.908 6.7 4"
              />
              <path
                d="M12 20l4 -9l4 9"
              />
              <path
                d="M19.1 18h-6.2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
